packages/webapp/src/components/Finances/Transaction/Mobile/List.jsx

```diff
@@ -38,9 +40,19 @@ export const MainContent = ({ t, note, typeLabel, amount, icon, currencySymbol }
 const Rows = ({ t, data, currencySymbol }) => {
   const { expandedIds, toggleExpanded } = useExpandable({ isSingleExpandable: true });
   const rows = [];
+  let groupDate = null;
 
   data.forEach((values, index) => {
-    // TODO: LF-3746 add "date" row as necessary
+    const itemDate = new Date(values.date);
+
+    if (!isSameDay(groupDate, itemDate)) {
+      groupDate = itemDate;
+      rows.push(
+        <div key={values.date} className={styles.transactionDate}>
```

Comment:
I'd suggest we use a `h2` heading tag here to make it more semantic, since this is sort of a "title" for the subsequent transactions

Reply:
Thank you, I will update it in my next PR!